Modules/StudyProgramme/classes/class.ilObjStudyProgrammeAutoMembershipsGUI.php

```diff
@@ -238,32 +264,98 @@ protected function save()
         $this->getObject()->storeAutomaticMembershipSource($src_type, (int) $src_id);
     }
 
-    /**
-     * Delete entries.
-     */
-    protected function delete()
+    protected function deleteConfirmation()
     {
         $post = $_POST;
         $field = self::CHECKBOX_SOURCE_IDS;
-        if (array_key_exists($field, $post)) {
-            foreach ($post[$field] as $src_id) {
-                list($type, $id) = explode('-', $src_id);
-                $this->getObject()->deleteAutomaticMembershipSource((string) $type, (int) $id);
-            }
+
+        if (!array_key_exists($field, $post)) {
+            ilUtil::sendInfo($this->lng->txt('prg_delete_nothing_selected'), true);
+            $this->ctrl->redirect($this, self::CMD_VIEW);
+        }
+
+        $str = '';
+        foreach ($post[$field] as $data) {
+            $str .= $data . ' ';
         }
+
+        $str = base64_encode($str);
+        $message = $this->lng->txt('prg_delete_confirmation');
+        $this->ctrl->setParameterByClass(self::class, 'type_ids', $str);
+        $delete = $this->ctrl->getFormActionByClass(self::class, self::CMD_DELETE);
+        $cancel = $this->ctrl->getFormActionByClass(self::class, self::CMD_VIEW);
+        $this->ctrl->clearParameterByClass(self::class, 'type_ids');
+
+        $buttons = [
+            $this->button_factory->standard($this->lng->txt('prg_confirm_delete'), $delete),
+            $this->button_factory->standard($this->lng->txt('prg_cancel'), $cancel)
+        ];
+
+        $message_box = $this->message_box_factory->confirmation($message)->withButtons($buttons);
+
+        $this->tpl->setContent($this->ui_renderer->render($message_box));
+    }
+
+    protected function delete()
+    {
+        $get = $_GET;
```

Comment:
We already have the request. Let's use it here.

Reply:
done